src/riak_kv_index_hashtree.erl

```diff
@@ -1029,3 +1039,8 @@ maybe_get_vnode_lock(SrcPartition, Pid) ->
         false ->
             ok
     end.
+
+maybe_callback(undefined) ->
+    ok;
+maybe_callback(Callback) ->
+    Callback().
```

Comment:
After discussion w/ @JeetKunDoug @fadushin... look at https://github.com/basho/riak_kv/pull/1361/files#r55219339 for cleanup and where to move the try/catch.